src/main/java/com/example/finance/service/BudgetService.java

```diff
@@ -0,0 +1,83 @@
+package com.example.finance.service;
+
+import com.example.finance.exception.model.BackendException;
+import com.example.finance.mapper.BudgetMapper;
+import com.example.finance.model.dto.BudgetDto;
+import com.example.finance.model.entity.BudgetEntity;
+import com.example.finance.model.entity.CategoryEntity;
+import com.example.finance.model.entity.UserAccountEntity;
+import com.example.finance.repository.BudgetRepository;
+import com.example.finance.repository.CategoriesRepository;
+import com.example.finance.repository.UserAccountRepository;
+import com.example.finance.utils.MessageConstants;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.UUID;
+
+@AllArgsConstructor
+@Service
+@Slf4j
+public class BudgetService {
+
+    private final String BUDGET_NOT_FOUND_EXCEPTION_MESSAGE = "No budget found for id: ";
+
+    private final BudgetRepository budgetRepository;
+    private final CategoriesRepository categoriesRepository;
+    private final UserAccountRepository userAccountRepository;
+    private final BudgetMapper budgetMapper;
+
+    public List<BudgetDto> getByUserId(UUID userId) {
+        return toDtoList(budgetRepository.findByUserAccountEntityUserId(userId));
+    }
+
+    public BudgetDto getById(UUID id) {
+        return budgetRepository.findById(id)
+                .map(budgetMapper::toDto)
+                .orElseThrow(() -> new BackendException(BUDGET_NOT_FOUND_EXCEPTION_MESSAGE + id));
+    }
+
+
+    public BudgetDto create(BudgetDto budgetDto) {
+        UserAccountEntity userAccountEntity = userAccountRepository.findById(budgetDto.userId())
+                .orElseThrow(() -> new BackendException(MessageConstants.USER_NOT_FOUND));
+        CategoryEntity categoryEntity = categoriesRepository.findById(budgetDto.categoryId())
+                .orElseThrow(() -> new BackendException(MessageConstants.CATEGORY_NOT_FOUND));
+        BudgetEntity budgetEntity = budgetMapper.toEntity(budgetDto);
+        budgetEntity.setUserAccountEntity(userAccountEntity);
+        budgetEntity.setCategoryEntity(categoryEntity);
+        budgetRepository.save(budgetEntity);
+        log.debug(String.format("Created Budget with id %s and amount %s",
+                budgetEntity.getBudgetId(),
+                budgetEntity.getAmount()));
+        return budgetMapper.toDto(budgetEntity);
```

Comment:
If you are returning some data to client after save, use the entity which `.save()` method will return to you :)

Reply:
Fix it in all places